### PR TITLE
Remove hot restart leftovers

### LINKED ISSUE #19043: Hot restart naming leftovers
- Cluster property `hazelcast.hotrestart.free.native.memory.percentage` was deprecated in 4.2. Should it be removed in 5.0?
- Should public service/package `HotRestartService` / `com.hazelcast.hotrestart` be renamed for 5.0? If so, it would break public API.
- `RestEndpointGroup#HOT_RESTART`

### PR DESCRIPTION
**Pull request by ufukyilmaz**
This pr aims to remove the public leftovers of the hotrestart to
persistence renaming.
While doing this, I deprecate the `HotRestartService` and didn't make
many changes to it. The only change in `HotRestartService` is that
the package name of the `BackupState` class which hot restart service
depends on has changed. 

Also, I made the `HotRestartService` extends `PersistenceService`, so
gets the persistence service by just casting it here:
https://github.com/hazelcast/hazelcast/pull/19404/files#diff-5e75745e3690f9d2b5833b1ee7d3d7e06e6adefbfd8d9c78c28bde19b3fc6915R346-R352 

Breaking changes: 

- Renamed `c.h.hotrestart.BackupTask`->  `c.h.persistence.BackupTask` 
- Renamed `c.h.hotrestart.BackupTaskState` -> `c.h.persistence.BackupTaskState`
- Added `boolean isBackupEnabled()` to `PersistenceService`/`HotRestartService` that does
not have a default implementation. (I don't expect users to extend this `HotRestartService`
and a default implementation that depends on the deprecated method is looking dirty for me)

Tasks:
- [x] Remove cluster property `hazelcast.hotrestart.free.native.memory.percentage` was deprecated in 4.2.
- [x] Deprecate `HotRestartService` and create `PersistenceService` for replacing this functionality. Move other classes in the `com.hazelcast.hotrestart` package into `com.hazelcast.persistence`
- [x] Add `RestEndpointGroup#PERSISTENCE` group name that made it alias to `RestEndpointGroup#HOT_RESTART`. 
Deprecated this `RestEndpointGroup#HOT_RESTART` enum and started to use the newly added `PERSISTENCE` for representing this group while listing the enabled endpoint groups and so on.
- [x] Deprecated the endpoint including `hot` staff and add their alternatives that don't include the `hot` prefix. 

`URI_CLUSTER_MANAGEMENT_BASE_URL/hotBackup` ->`URI_CLUSTER_MANAGEMENT_BASE_URL/backup`
`URI_CLUSTER_MANAGEMENT_BASE_URL/hotBackupInterrupt` -> `URI_CLUSTER_MANAGEMENT_BASE_URL/backupInterrupt`

Missing staff: I couldn't manage to add deprecation notice to these deprecated API's responses yet. 

Fixes #19043
EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/4241

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/persistence/PersistenceService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.persistence;
+
+/**
+ * Service for interacting with Hot Restart. For example - starting cluster-wide hot restart data backups, determining the
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/persistence/PersistenceService.java:20]: Need to eliminate "hot restart"/"hot backup" references from the javadoc as well eg

```suggestion
 * Service for interacting with Persistence. For example - starting cluster-wide data backups, determining the
```

File: hazelcast/src/main/java/com/hazelcast/instance/impl/NodeExtension.java [hunk 2]
```diff
@@ -337,9 +338,18 @@ default boolean isClientFailoverSupported() {
      */
     boolean registerListener(Object listener);
 
-    /** Returns the public hot restart service */
+    /**
+     * Returns the public hot restart service
```
[the remark this answers is not in the document]
Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/instance/impl/NodeExtension.java:342]: I agree with this comment 👍 . I will remove the added `NodeExtension#getPersistenceService` staff.

File: hazelcast/src/main/java/com/hazelcast/internal/monitor/HotRestartState.java [hunk 3]
```diff
@@ -16,8 +16,8 @@
 
 package com.hazelcast.internal.monitor;
 
-import com.hazelcast.hotrestart.BackupTaskStatus;
 import com.hazelcast.json.internal.JsonSerializable;
+import com.hazelcast.persistence.BackupTaskStatus;
```
[the remark this answers is not in the document]
Comment (emre-aydin) [reply hazelcast/src/main/java/com/hazelcast/internal/monitor/HotRestartState.java:20]: nope, we have our own counterparts for them.

**Comment by ufukyilmaz** [comment]
run-lab-run

**Comment by ufukyilmaz** [comment]
run-lab-run

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/config/ConfigXmlGeneratorTest.java [hunk 4]
```diff
@@ -1991,8 +1991,37 @@ public void testAllEnabledRestApiConfig() {
...
+        restApiConfig.enableGroups(RestEndpointGroup.CLUSTER_READ, RestEndpointGroup.HEALTH_CHECK,
+                RestEndpointGroup.HOT_RESTART, RestEndpointGroup.WAN);
+        restApiConfig.disableGroups(RestEndpointGroup.CLUSTER_WRITE, RestEndpointGroup.DATA);
+        Config config = new Config();
+        config.getNetworkConfig().setRestApiConfig(restApiConfig);
+        RestApiConfig generatedConfig = getNewConfigViaXMLGenerator(config).getNetworkConfig().getRestApiConfig();
+        assertTrue(generatedConfig.toString() + " should be compatible with " + restApiConfig.toString(),
+                new ConfigCompatibilityChecker.RestApiConfigChecker().check(restApiConfig, generatedConfig));
+    }
+
```
Comment (mdumandag) [inline hazelcast/src/test/java/com/hazelcast/config/ConfigXmlGeneratorTest.java:2003]: Minor: Unnecessary newline

**Review by mdumandag — APPROVED** [review]
Approving, everything seems OK from the client's perspective. Also, left some minor comments

**Review by petrpleshachkov — APPROVED** [review]
Looks good to me!

**Comment by ufukyilmaz** [comment]
Thanks for your reviews @vbekiaris @mdumandag @petrpleshachkov

**Pull request merged**